app/src/pages/playground/PlaygroundTemplate.tsx

```diff
@@ -26,18 +27,37 @@ interface PlaygroundTemplateProps extends PlaygroundInstanceProps {}
 export function PlaygroundTemplate(props: PlaygroundTemplateProps) {
   const [dialog, setDialog] = useState<React.ReactNode>(null);
   const instanceId = props.playgroundInstanceId;
+  const updateInstance = usePlaygroundContext((state) => state.updateInstance);
   const instances = usePlaygroundContext((state) => state.instances);
   const instance = instances.find((instance) => instance.id === instanceId);
   const index = instances.findIndex((instance) => instance.id === instanceId);
   const prompt = instance?.prompt;
   const promptId = prompt?.id;
-  const updateInstancePrompt = usePlaygroundContext(
-    (state) => state.updateInstancePrompt
-  );
 
-  // TODO(apowell): Sync instance state with promptId + version (or latest if unset)
-  // If it exists, and we can fetch it from gql, replace the instance with it
-  // If it doesn't exist, or we can't fetch it from gql, set the promptId to null
+  const onChangePrompt = useCallback(
+    async (promptId: string | null) => {
+      if (!promptId) {
+        updateInstance({
+          instanceId,
+          patch: {
+            prompt: null,
+          },
+        });
+        return;
+      }
+
+      const newInstance = await fetchPlaygroundPromptAsInstance(promptId);
+      if (newInstance) {
+        updateInstance({
+          instanceId,
+          patch: {
+            ...newInstance,
+          },
+        });
```

Comment:
nice, I think this is clean and leaves room for other things to happen on this action.